src/runtime-rs/crates/hypervisor/src/device/driver/vfio.rs

```diff
@@ -266,15 +266,17 @@ impl VfioDevice {
         let dev_type = dev_info.dev_type.as_str();
         let driver_type = VfioBusMode::driver_type(dev_type).to_owned();
 
-        Self {
+        let mut vfio_device = Self {
             device_id,
             attach_count: 0,
             bus_mode: VfioBusMode::PCI,
             driver_type,
             config: dev_info.clone(),
             devices,
             device_options,
-        }
+        };
+
+        vfio_device.set_vfio_device().cloned()
```

Comment:
Is the cloning necessary here?  If I understand correctly, `vfio_device` is local to this function so do we mind if it gets moved from into the return value?

Reply:
fixed